Diagnostic controls don't seem to work as expected for linked diagnostic groups

### Description

IIUC the [design doc](https://github.com/swiftlang/swift-evolution/blob/f248f47e880fdab08a35ff681403cf93a1c84445/proposals/0443-warning-control-flags.md#diagnostic-groups) implies that specifying diagnostic flags for a parent group should apply them to children unless the children are explicitly passed with different values. this does not appear to work for the `RegionIsolation` (parent) and `SendingRisksDataRace` (child) groups.

### Reproduction

```swift
// swiftc -language-mode 5 -strict-concurrency=complete -Werror RegionIsolation
class C {}

func send(_: @escaping @Sendable () -> Void) {}

func f(_ c: C) {
    send { _ = c } // ⚠️ still a warning despite upgrading parent diagnostic group to an error
}
```


### Expected behavior

the parent group diagnostic settings should be inherited by child groups

### Environment

Swift version 6.3-dev (LLVM 4f7a3dbb7e97ac4, Swift 477d32c08627ddd)
Target: x86_64-unknown-linux-gnu

// 6.3 dev snapshot

### Additional information

godbolt: https://swift.godbolt.org/z/EcxW8fKxe

This reproduction has a different diagnostic now, hmm.
```
warning: capture of 'c' with non-Sendable type 'C' in a '@Sendable' closure; this is an error in the Swift 6 language mode [#SendableClosureCaptures]
1 | class C {}
  |       `- note: class 'C' does not conform to the 'Sendable' protocol
2 |
3 | func send(_: @escaping @Sendable () -> Void) {}
4 |
5 | func f(_ c: C) {
6 |     send { _ = c } // ⚠ still a warning despite upgrading parent diagnostic group to an error
  |                `- warning: capture of 'c' with non-Sendable type 'C' in a '@Sendable' closure; this is an error in the Swift 6 language mode [#SendableClosureCaptures]
7 | }
```

This still triggers an RBI error:

```swift
class NS {}

@MainActor func take(_ x: NS) {}

func test() async {
    let x = NS()
    await take(x)   // x sent into the main actor's region
    print(x)        // used after sending
}
```

But I think the issues with Werror for child groups is fixed:

```
❮ swiftc -language-mode 5 -strict-concurrency=complete -Werror RegionIsolation rbi_repro.swift
rbi_repro.swift:7:11: error: sending 'x' risks causing data races; this is an error in the Swift 6 language mode [#RegionIsolation::SendingRisksDataRace]
 5 | func test() async {
 6 |     let x = NS()
 7 |     await take(x)   // x sent into the main actor's region
   |           |- error: sending 'x' risks causing data races; this is an error in the Swift 6 language mode [#RegionIsolation::SendingRisksDataRace]
   |           `- note: sending 'x' to main actor-isolated global function 'take' risks causing data races between main actor-isolated and local nonisolated uses
 8 |     print(x)        // used after sending
   |           `- note: access can happen concurrently
 9 | }
10 |

[#RegionIsolation]: <https://docs.swift.org/compiler/documentation/diagnostics/region-isolation>
[#SendingRisksDataRace]: <https://docs.swift.org/compiler/documentation/diagnostics/sending-risks-data-race>
❮ swiftc -language-mode 5 -strict-concurrency=complete  rbi_repro.swift
rbi_repro.swift:7:11: warning: sending 'x' risks causing data races; this is an error in the Swift 6 language mode [#RegionIsolation::SendingRisksDataRace]
 5 | func test() async {
 6 |     let x = NS()
 7 |     await take(x)   // x sent into the main actor's region
   |           |- warning: sending 'x' risks causing data races; this is an error in the Swift 6 language mode [#RegionIsolation::SendingRisksDataRace]
   |           `- note: sending 'x' to main actor-isolated global function 'take' risks causing data races between main actor-isolated and local nonisolated uses
 8 |     print(x)        // used after sending
   |           `- note: access can happen concurrently
 9 | }
10 |

[#RegionIsolation]: <https://docs.swift.org/compiler/documentation/diagnostics/region-isolation>
[#SendingRisksDataRace]: <https://docs.swift.org/compiler/documentation/diagnostics/sending-risks-data-race>
```

I'm going to leave this open to track adding a test to make sure we don't regress this / checking that this is already tested

--- a/test/diagnostics/werror_parent_group_region_isolation.swift
+++ b/test/diagnostics/werror_parent_group_region_isolation.swift
@@ -1,0 +1,22 @@
+// Default: the child group's diagnostic is a warning in language mode 5.
+// RUN: %target-swift-frontend -emit-sil -o /dev/null %s -target %target-swift-5.1-abi-triple -strict-concurrency=complete -swift-version 5 -verify -verify-additional-prefix warnonly-
+
+// Ensure escalated with `-Werror` on the child group.
+// RUN: %target-swift-frontend -emit-sil -o /dev/null %s -target %target-swift-5.1-abi-triple -strict-concurrency=complete -swift-version 5 -Werror SendingRisksDataRace -verify -verify-additional-prefix erronly-
+
+// Ensure escalated with `-Werror` on the parent group.
+// RUN: %target-swift-frontend -emit-sil -o /dev/null %s -target %target-swift-5.1-abi-triple -strict-concurrency=complete -swift-version 5 -Werror RegionIsolation -verify -verify-additional-prefix erronly-
+
+// REQUIRES: concurrency
+
+class NS {}
+
+@MainActor func take(_ x: NS) {}
+
+func test() async {
+  let x = NS()
+  await take(x) // expected-warnonly-warning {{sending 'x' risks causing data races}}
+  // expected-erronly-error@-1 {{sending 'x' risks causing data races}}
+  // expected-note@-2 {{sending 'x' to main actor-isolated global function 'take' risks causing data races between main actor-isolated and local nonisolated uses}}
+  print(x) // expected-note {{access can happen concurrently}}
+}

